test/tools/cli/IonProcess.java

```diff
@@ -0,0 +1,312 @@
+                                      IonWriter ionWriterForErrorReport,
+                                      IonReader ionReader,
+                                      SymbolTable curTable) throws IOException {
+        while (ionReader.next() != null) {
+            if (!isSameSymbolTable(ionReader.getSymbolTable(), curTable)) {
+                curTable = ionReader.getSymbolTable();
+                ImportDescriptor imports[] = symbolTableToImports(curTable.getImportedTables());
+                new Event(EventType.SYMBOL_TABLE, null, null, null,
+                        null, null, imports, 0 )
+                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport);
+            }
+
+            if (IonType.isContainer(ionReader.getType())) {
+                //get current Ion type and depth
+                IonType curType = ionReader.getType();
+                int curDepth = ionReader.getDepth();
+
+                //write a Container_Start event and step in
+                ionStreamToEvent(EventType.CONTAINER_START, ionReader)
+                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport);
+                ionReader.stepIn();
+
+                //recursive call
+                processEvent(ionWriterForOutput, ionWriterForErrorReport, ionReader, curTable);
+
+                //write a Container_End event and step out
+                new Event(EventType.CONTAINER_END, curType, null, null,
+                         null, null, null, curDepth )
+                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport);
+                ionReader.stepOut();
+            } else {
+                ionStreamToEvent(EventType.SCALAR, ionReader).writeOutput(ionWriterForOutput, ionWriterForErrorReport);
+            }
+        }
+    }
+
+    /**
+     *  This function creates a new file for output stream with the fileName, If file name is empty or undefined,
+     *  it will create a default output which are System.out or System.err.
+     */
+    private static BufferedOutputStream initOutputStream(String fileName, String defaultValue) {
+        BufferedOutputStream outputStream = null;
+        try {
+            if (fileName != null && fileName.length() != 0) {
+                File myFile = new File(fileName);
+                FileOutputStream out = new FileOutputStream(myFile);
+                outputStream = new BufferedOutputStream(out, BUFFER_SIZE);
+            } else {
+                switch (defaultValue){
+                    case SYSTEM_OUT_DEFAULT_VALUE:
+                        outputStream = new BufferedOutputStream(System.out, BUFFER_SIZE);
+                        break;
+                    case SYSTEM_ERR_DEFAULT_VALUE:
+                        outputStream = new BufferedOutputStream(System.err, BUFFER_SIZE);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        } catch (IOException e ) {
```

Comment:
I recommend removing the try/catch from this method and moving the invocations of this method into the try-with-resources of the caller, which already handles IOException.

Reply:
moved